linkerd/core/src/main/scala/io/buoyant/linkerd/RequestEvaluator.scala

```diff
@@ -0,0 +1,168 @@
+package io.buoyant.linkerd
+
+import com.twitter.finagle._
+import com.twitter.finagle.client.Transporter.EndpointAddr
+import com.twitter.finagle.http._
+import com.twitter.finagle.naming.buoyant.DstBindingFactory
+import com.twitter.util._
+import io.buoyant.namer.DelegateTree._
+import io.buoyant.namer.{DelegateTree, Delegator}
+import io.buoyant.router.RouterLabel
+import io.buoyant.router.RoutingFactory
+import io.buoyant.router.RoutingFactory.BaseDtab
+import io.buoyant.router.context.{DstBoundCtx, DstPathCtx}
+
+class RequestEvaluator(
```

Comment:
We should think about what we want to call this.  I don't think Evaluator is quite right.  This is a kind of tracing but we should be careful not to conflate it with the type of tracing already in Linkerd (zipkin etc).  

Perhaps a name related to ActiveTracing (as opposed to the "passive" tracing that zipkin does).  Or maybe RoutingInfoFilter... hmmm...

Reply:
I liked the name you mentioned earlier, RequestTracer. We could call it RequestActiveTracer (RAT haha) so we don't confuse it with passive tracing.